contracts/RelayerHub.sol

```diff
@@ -8,88 +8,193 @@ import "./System.sol";
+
+        address relayerAddress = managersAndRelayers[managerAddress];
+
+        delete (relayManagersExistMap[managerAddress]);
+        delete (managersAndRelayers[managerAddress]);
+
+        manager memory a = managers[managerAddress];
+        managerAddress.transfer(a.deposit.sub(a.dues));
+        address payable systemPayable = address(uint160(SYSTEM_REWARD_ADDR));
+        systemPayable.transfer(a.dues);
+
+        delete (managers[managerAddress]);
+        delete (managersRegistered[managerAddress]);
+
+        // emit success event
+        emit removeManagerByGovEvent(managerAddress);
+        if (relayerAddress != address(0)) {
+            emit removeRelayerEvent(relayerAddress);
+        }
+    }
+
+    function addManagerByGov(address managerToBeAdded) internal {
+        require(!relayManagersExistMap[managerToBeAdded], "manager already exists");
+        require(!isContract(managerToBeAdded), "contract is not allowed to be a manager");
+
+        relayManagersExistMap[managerToBeAdded] = true;
+
+        emit addManagerByGovEvent(managerToBeAdded);
+    }
+
+    function registerManager() internal onlyNonRegisteredManager {
+        require(msg.value == requiredDeposit, "deposit value is not exactly the same");
+        managers[msg.sender] = manager(requiredDeposit, dues);
+        managersRegistered[msg.sender] = true;
+        emit registerManagerEvent(msg.sender);
+    }
+
+    function addRelayer(address relayerToBeAdded) internal onlyRegisteredManager noProxy {
+        require(!relayerExistsMap[relayerToBeAdded], "relayer already exists");
+        require(!isContract(relayerToBeAdded), "contract is not allowed to be a relayer");
+
+        managersAndRelayers[msg.sender] = relayerToBeAdded;
+        relayerExistsMap[relayerToBeAdded] = true;
+        emit editRelayerEvent(relayerToBeAdded);
+    }
+
+    function editRelayer(address relayerToBeAdded) external onlyRegisteredManager noProxy {
+        require(!relayerExistsMap[relayerToBeAdded], "relayer already exists");
+        require(!isContract(relayerToBeAdded), "contract is not allowed to be a relayer");
+
+        managersAndRelayers[msg.sender] = relayerToBeAdded;
+        relayerExistsMap[relayerToBeAdded] = true;
+        emit editRelayerEvent(relayerToBeAdded);
+    }
+
+    function registerManagerAddRelayer(address relayer) external payable onlyNonRegisteredManager {
+        registerManager();
+        addRelayer(relayer);
+    }
+
+    function removeRelayer() external onlyRegisteredManager {
+        if (managersAndRelayers[msg.sender] == address(0)) {
+            require(false, "relayer doesn't exist for this manager");
+        }
+
+        address r = managersAndRelayers[msg.sender];
+
+        delete (relayerExistsMap[r]);
+        delete (managersAndRelayers[msg.sender]);
+
+        emit removeRelayerEvent(r);
+    }
+
+    function isRelayer(address relayerAddress) external override view returns (bool){
```

Comment:
This function is never used inside the contract. I guess there are 2 options:

- Delete it and also remove the `IRelayerHub` interface from the contract declaration (I don't know what are the implications though)
- Use it in the other functions instead of directly using the `relayerExistsMap`